File: api/src/services/share_service.py
from utils import file as file_util
from constants.data_folder_name import DATA_FOLDER_NAME
class Share_Service:

    MONTHLY_DATA_FILE_NAME = "monthly.json"
    WEEKLY_DATA_FILE_NAME = "weekly.json"
    DAILY_DATA_FILE_NAME = "daily.json"
    QUATERLY_DATA_FILE_NAME = "quaterly.json"
    YEARLY_DATA_FILE_NAME = "yearly.json"

    FILE_PATH = "/home/serpant/Desktop/work/share_report/data/"

    def get_file_path_by_category(self, category): 
        return self.FILE_PATH + DATA_FOLDER_NAME[category]

    def fetch_monthly_data(self, category):
        file_path = self.get_file_path_by_category(category)
        return file_util.read_file(file_path + "/" + self.MONTHLY_DATA_FILE_NAME)

    def fetch_weekly_data(self, category):
        file_path = self.get_file_path_by_category(category)
        return file_util.read_file(file_path + "/" + self.WEEKLY_DATA_FILE_NAME)

    def fetch_daily_data(self, category):
        file_path = self.get_file_path_by_category(category)
        return file_util.read_file(file_path + "/" +self.DAILY_DATA_FILE_NAME)
    
    def fetch_quaterly_data(self, category):
        file_path = self.get_file_path_by_category(category)
        return file_util.read_file(file_path +"/" + self.QUATERLY_DATA_FILE_NAME)
    
    def fetch_yearly_data(self, category):
        file_path = self.get_file_path_by_category(category)
        return file_util.read_file(file_path +"/"+ self.YEARLY_DATA_FILE_NAME)


share_services = Share_Service()

File: app/src/services/shareService.js
import CONFIG from '../config';
import * as http from '../utils/http';

export const fetchDailyData = async (shareSymbol, name) => {
  const { data } = await http.get(
    `${CONFIG.END_POINTS.shareinfo}/d?shareSymbol=${shareSymbol}&shareName=${name}`
  );
  return data;
};

export async function fetchWeeklyData(shareSymbol, name) {
  const { data } = await http.get(
    `${CONFIG.END_POINTS.shareinfo}/w?shareSymbol=${shareSymbol}&shareName=${name}`
  );

  return data;
}

export async function fetchMonthlyData(shareSymbol, name) {
  const { data } = await http.get(
    `${CONFIG.END_POINTS.shareinfo}/m?shareSymbol=${shareSymbol}&shareName=${name}`
  );

  return data;
}

export async function fetchQuaterData(shareSymbol, name) {
  const { data } = await http.get(
    `${CONFIG.END_POINTS.shareinfo}/q?shareSymbol=${shareSymbol}&shareName=${name}`
  );

  return data;
}

export async function fetchYearlyData(shareSymbol, name) {
  const { data } = await http.get(
    `${CONFIG.END_POINTS.shareinfo}/y?shareSymbol=${shareSymbol}&shareName=${name}`
  );

  return data;
}

const shareServices = {
  fetchDailyData,
  fetchWeeklyData,
  fetchMonthlyData,
  fetchQuaterData,
  fetchYearlyData,
};

export default shareServices;
